pkgs/development/tools/qtcreator/0001-Fix-clang-libcpp-regexp.patch

```diff
@@ -13,3 +13,12 @@ index b514c46..5f96358 100644
      static const QRegularExpression includeRegExp("\\A(" + cppIncludes + ")\\z");
  
      // The same as includeRegExp but also matches /usr/local/include
+@@ -134,8 +134,6 @@ void removeClangSystemHeaderPaths(HeaderPaths &headerPaths)
+ 
+ void HeaderPathFilter::tweakHeaderPaths()
+ {
+-    removeClangSystemHeaderPaths(builtInHeaderPaths);
```

Comment:
Can you explain this change? It doesn't look like a modification to a regex.

Reply:
I mentioned this in my previous comment.

> Compilation was successful, so i opened up test project in qtcreator only to see that clang code model fails to find clang includes.
After some code inspection i found out that qtcreator searches for includes in directory that is bundled with it, to make it so system clang don't interfere with search pathes there is method in code removeClangSystemHeaderPaths(builtInHeaderPaths);. In our case we want qtcreator to search for clang system include, so i removed it.

Reply:
And why is this combined with the already existing Fix-clang-libcpp-regex patch?

Reply:
Cause it's solves the same problem, i may split it into two different patches or rename patch if needed.

Reply:
It should be very obvious which patches are applied for which reason. Otherwise future maintainers won't be able to adapt or remove them. So adding an explanation to the comment and/or giving it a more descriptive name would be nice.

Reply:
Looking at the code [1], it appears that "removeClangSystemHeaderPaths" removes the system clang include directories. It's not clear at all why we should disable this. Perhaps it is supposed to be able to find includes from `clang_qt_vendor` instead?

[1] https://github.com/qt-creator/qt-creator/blob/master/src/plugins/cpptools/headerpathfilter.cpp

Reply:
Also is this an issue that is seen in 4.10.0 only or 4.9.1 as well?

Reply:
Ok, i've splitted up the patch into two different patches and added comment.

>> Looking at the code [1], it appears that "removeClangSystemHeaderPaths" removes the system clang include directories. It's not clear at all why we should disable this. Perhaps it is supposed to be able to find includes from clang_qt_vendor instead?
As i previously mentioned qtcreator in classic linux packaging comes with clang, clazy and it's includes in one single package. Path for it is something like `/usr/share/qtcreator/include ...`
In our case we reuse system clang recipe to build different sources for clang, so it's pathes still found by system paths.

>> Also is this an issue that is seen in 4.10.0 only or 4.9.1 as well?
They changed behavior, previously system paths was removed only if there is "darwin" in system triple, as i don't have mac to test it i leave'd it untouched. So 4.9.1 worked as intended.